playwright/sync_base.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from typing import Any, Callable, Union
+
+from playwright.wait_helper import WaitHelper
+
+loop = asyncio.get_event_loop()
+
+
+class Event:
+    def __init__(
+        self,
+        sync_base: "SyncBase",
+        event: str,
+        predicate: Callable[[Any], bool] = None,
+        timeout: int = None,
+    ) -> None:
+        self._value: Any = None
+
+        wait_helper = WaitHelper()
+        wait_helper.reject_on_timeout(
+            timeout or 30000, f'Timeout while waiting for event "${event}"'
+        )
+        self._future = asyncio.ensure_future(
+            wait_helper.wait_for_event(sync_base._async_obj, event, predicate)
+        )
+
+    @property
+    def data(self) -> Any:
```

Comment:
I would probably use `.value` instead since it seems more common in the Python world. Pytest is using `.value` e.g. with `with pytest.raises()`